### PR TITLE
Establish etcd/client APIs

### PR DESCRIPTION
**Pull request by bcwaldon**
This is in a reviewable state, but keep in mind that there is still a lot to be done to "finish" the etcd/client package.

### CODE REVIEW

**Comment by jonboulle** [comment]
@bcwaldon are you going to do a mega godoc commit later or should I harass you now?

**Comment by bcwaldon** [comment]
@jonboulle mega commit once everything settles out.

**Comment by jonboulle** [comment]
ok, you can bet I'm looking forward to it.

File: client/keys.go [hunk 1]
```diff
@@ -54,11 +62,34 @@ func NewDiscoveryKeysAPI(c HTTPClient) KeysAPI {
 }
 
 type KeysAPI interface {
-	Create(ctx context.Context, key, value string, ttl time.Duration) (*Response, error)
+	Set(ctx context.Context, key, value string, opts *SetOptions) (*Response, error)
+	Create(ctx context.Context, key, value string) (*Response, error)
```
Comment (yichengq) [inline client/keys.go]: do you have any plan how create will support ttl?

Comment (bcwaldon) [inline client/keys.go]: Create and Update don't get TTLs - use Set.

Comment (bcwaldon) [inline client/keys.go]: To expand on that - we're making a tradeoff here between readability and functionality. The common case is to Create/Update a key without a TTL. We don't want to force users to set a dummy value for the TTL, then have to question what it means every time they read the code. For example, what do the `-1` or `0` mean below?

```
Create(ctx, "/foo", "bar", -1)
Update(ctx, "/foo", "baz", 0)
```

If you care about setting the TTL or doing any comparisons, define them in a SetOptions struct and use the Set method. You could do a Create with a TTL through Set with `SetOptions{PrevExist: false, TTL: time.Minute}`

Comment (yichengq) [inline client/keys.go]: understand. `Create` and `Update` just does the minimal work, and `Set` gives the full features.

Comment (bcwaldon) [inline client/keys.go]: You got it.

**Comment by bcwaldon** [comment]
Moved to https://github.com/coreos/etcd/pull/2200

**Pull request closed without merging**